ibazel/live_reload/server.go

```diff
@@ -82,25 +59,47 @@ func (l *LiveReloadServer) TargetDecider(rule *blaze_query.Rule) {
 		}
 	}
 }
-func contains(l []string, e string) bool {
-	for _, i := range l {
-		if i == e {
-			return true
-		}
-	}
-	return false
+
+func (l *LiveReloadServer) ChangeDetected(c lifecycle.Lifecycle, targets []string, changeType string, change string) {
 }
 
-func (l *LiveReloadServer) ChangeDetected(changeType string) {}
+func (l *LiveReloadServer) BeforeCommand(c lifecycle.Lifecycle, targets []string, command string) {}
 
-func (l *LiveReloadServer) BeforeCommand(command string) {}
-func (l *LiveReloadServer) AfterCommand(command string, success bool) {
+func (l *LiveReloadServer) AfterCommand(c lifecycle.Lifecycle, targets []string, command string, success bool) {
 	if l.lrserver != nil {
 		fmt.Fprintf(os.Stderr, "Triggering live reload\n")
 		l.lrserver.Reload("reload")
+		c.ReloadTriggered(c, targets)
```

Comment:
That's the tricky bit. Unless the the profiler is aware of whether or not livereload is enabled (`if l.lrserver != nil` above) then it can't know if a livereload was triggered. Hence the modifications to allow for lifecycle events to create other lifecycle events by having them take a Lifecycle interface that can be calle.

Reply:
I'm open to other suggestions. Presently the profiler is using the livereload event to know if it should accept REMOTE_EVENT from the browser. If live reload is disabled then it ignore the REMOTE_EVENTs as they would be user driven and not useful to profiling the RTT.